dash/include/dash/algorithm/Transform.h

```diff
@@ -271,40 +296,65 @@ GlobOutputIt transform(
   BinaryOperation binary_op)
 {
   DASH_LOG_DEBUG("dash::transform(af, al, bf, outf, binop)");
+  auto &pattern = out_first.pattern();
   // Outut range different from rhs input range is not supported yet
-  auto in_first = in_a_first;
-  auto in_last  = in_a_last;
-  std::vector<ValueType> in_range;
+  ValueType* in_first = &(*in_a_first);
+  ValueType* in_last  = &(*in_a_last);
+  // Number of elements in local range:
+  size_t num_local_elements     = std::distance(in_first, in_last);
+  auto out_last                 = out_first + num_local_elements;
+  if (out_last.gpos() > pattern.size()) {
+    DASH_THROW(dash::exception::OutOfRange,
+      "Too many input elements in dash::transform");
+  }
   if (in_b_first == out_first) {
     // Output range is rhs input range: C += A
     // Input is (in_a_first, in_a_last).
   } else {
     // Output range different from rhs input range: C = A+B
     // Input is (in_a_first, in_a_last) + (in_b_first, in_b_last):
-    std::transform(
-      in_a_first, in_a_last,
+    dash::copy(
       in_b_first,
-      std::back_inserter(in_range),
-      binary_op);
-    in_first = in_range.data();
-    in_last  = in_first + in_range.size();
+      in_b_first + std::distance(in_a_first, in_a_last),
+      out_first);
   }
 
   dash::util::Trace trace("transform");
 
-  // Resolve local range from global range:
-  // Number of elements in local range:
-  size_t num_local_elements     = std::distance(in_first, in_last);
   // Global iterator to dart_gptr_t:
   dart_gptr_t dest_gptr         = out_first.dart_gptr();
   // Send accumulate message:
-  trace.enter_state("transform_blocking");
-  dash::internal::transform_blocking_impl(
+  auto &team    = pattern.team();
+  size_t towrite = num_local_elements;
+  auto out_it    = out_first;
+  auto in_it     = in_first;
+  while (towrite > 0) {
+    auto   lpos  = out_it.lpos();
+    size_t lsize = pattern.local_size(lpos.unit);
```

Comment:
That is what I already mentioned multiple times. This approach only works for a single  continuous index range. Hence it would be better to use the new range based views interface by @fuchsto . There should be something like `dash::chunks(dash::local(...))`. Unfortunately I do not have detailed knowledge in this API either.